File: unit11-hw/ruby/Duck.rb
# Duck.rb | Defines the Duck class which inherits attributes and methods
#   from the Bird superclass.
#
# Begun by: Dr. Jump for CS 315 at King's College.
# Completed by: Robert Lowman
# Date: 11/29/16
####################################################

require_relative 'FlyingBird.rb'

class Duck < FlyingBird

  def call
    'Quack!'
  end

end

File: unit11-hw/python/Duck.py
# Duck.py | Defines the Duck class which inherits attributes and methods
#   from the Bird superclass.
#
# Completed by: Robert Lowman
# Date: 12/6/16
####################################################
import FlyingBird

class Duck(FlyingBird.FlyingBird):

    def __init__(self, name):
        FlyingBird.__init__(name)
        self.name = name

    def call(self):
        return 'Quack!'

    def className(self):
        return 'Duck'
